### PR TITLE
env variables for ACTION_HISTORY limits

### PR DESCRIPTION
**Pull request by fflorent**
## Context

Partial fix of https://github.com/gristlabs/grist-core/issues/1121

As a self-hoster, I want to be able to parameterize ACTION_HISTORY max sizes (both as rows and as disk space)

## Proposed solution

Introduce [2 env variables](https://github.com/gristlabs/grist-core/pull/1262/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R265-R266) to make those limits configurable.

I made these two values require to be strictly positive.

Also I introduced check for minimum values (`minValue`) and for the maximum values (`maxValue`) for `AppSettings.readInt()` / `AppSettings.requireInt()` and took the opportunity to force the timeout for OIDC to be positive.

Also as I introduced some changes in AppSettings, I started testing this module. I felt like it would take some time to test everything, so I start small with this test and only test properly the methods related to integers (`readInt` and `requireInt`).

## Related issues

Fixes https://github.com/gristlabs/grist-core/issues/1121

## Has this been tested?

- [x] 👍 yes, I added tests to the test suite
- [ ] 💭 no, because this PR is a draft and still needs work
- [ ] 🙅 no, because this is not relevant here
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -262,6 +262,8 @@ Grist can be configured in many ways. Here are the main environment variables it
...
 | APP_STATIC_INCLUDE_CUSTOM_CSS      | set to "true" to include custom.css (from APP_STATIC_URL) in static pages                                                                                                                                                                                                                                                                                     |
 | APP_UNTRUSTED_URL                  | URL at which to serve/expect plugin content.                                                                                                                                                                                                                                                                                                                  |
+| GRIST_ACTION_HISTORY_MAX_ROWS | Maximum number of rows allowed in ActionHistory before pruning (up to a 1.25 grace factor). Defaults to 1000.   |
+| GRIST_ACTION_HISTORY_MAX_BYTES | Maximum number of rows allowed in ActionHistory before pruning (up to a 1.25 grace factor). Defaults to 1.25Gb.   |
```
Comment (vviers) [inline README.md:266]:
```suggestion
| GRIST_ACTION_HISTORY_MAX_BYTES | Maximum number of rows allowed in ActionHistory before pruning (up to a 1.25 grace factor). Defaults to 1Gb.   |
```

File: app/server/lib/ActionHistoryImpl.ts [hunk 2]
```diff
@@ -12,14 +12,28 @@ import mapValues = require('lodash/mapValues');
...
+const ACTION_HISTORY_MAX_ROWS = section.flag('maxRows').requireInt({
+  envVar: 'GRIST_ACTION_HISTORY_MAX_ROWS',
+  defaultValue: 1000,
+
+  minValue: 1,
+});
+
+const ACTION_HISTORY_MAX_BYTES = section.flag('maxBytes').requireInt({
+  envVar: 'GRIST_ACTION_HISTORY_MAX_BYTES',
+  defaultValue: 1_000_0000_0000, // 1 GB.
```
Comment (vviers) [inline app/server/lib/ActionHistoryImpl.ts:32]:
```suggestion
  defaultValue: 1_000_000_000, // 1 GB.
```

**Review by vviers — APPROVED** [review]
LGTM except for 1 typo

**Review by hexaltation — CHANGES_REQUESTED** [review]
Please correct the value of 100GB to 1GB as suggested by @vviers 
for GRIST_ACTION_HISTORY_MAX_BYTES

Thank you for the tests :)

File: app/server/lib/ActionHistoryImpl.ts (hunk 2 above)
Comment (fflorent) [reply app/server/lib/ActionHistoryImpl.ts]: I instead used this notation which help avoiding the same issue in the future: `1e9`

**Review by paulfitz — APPROVED** [review]
Excellent, thanks, and especially appreciate the AppSettings tests.

**Pull request merged**